improve ci docs (#150)
pyroscope-io/docs: docs/ci.mdx

```

> (feel free to download and inspect the script's content, and also pin to a fixed commit)

Reviewer: ```suggestion
We recommend downloading and inspecting the script's content and then pinning to a fixed commit
```

Author: Don't actually recommend doing that, it's just in case someone doesn't want to run from `latest` due to security reasons or something.

> It will download the binary locally, which can be assed via `./pyroscope-ci`.

Reviewer: ```suggestion
It will download the binary locally, which can be accessed via `./pyroscope-ci`.
```

> If you want to download a specific version, first download the script and invoke it with `./install.sh $DESIRED_VERSION`

Reviewer: I think this is a little too much information and we could remove this line to keep things simple

Author: I think this is pretty important actually, otherwise people would always be running the latest version and therefore we could break their usage.
